src/pathology-report/ReportComponent.tsx

```diff
@@ -1,40 +1,476 @@
+
+  const referralStatusChange = (targetedElem, encounterInfo) => {
+    tempEncList.current = cloneDeep(encountersList);
+    if (!encounterInfo.referralStatusObs) {
+      postReferralStatusChangeObs(targetedElem, encounterInfo).then(
+        (response) => {
+          if (response.ok) {
+            const encIndex = tempEncList.current.findIndex(
+              (enc) => enc.encounterUuid == encounterInfo.encounterUuid
+            );
+            tempEncList.current[encIndex].referralStatusObs =
+              targetedElem.display;
+            tempEncList.current[encIndex].referralStatusObsUuid =
+              response.data.uuid;
+            setEncountersList(tempEncList.current);
+          } else {
+            setEncountersList(tempEncList.current);
+            //Need error message
+          }
+        }
+      );
+    } else if (encounterInfo.referralStatusObs) {
+      updateReferralStatusChangeObs(targetedElem, encounterInfo).then(
+        (response) => {
+          if (response.ok) {
+            const encIndex = tempEncList.current.findIndex(
+              (enc) => enc.encounterUuid == encounterInfo.encounterUuid
+            );
+            tempEncList.current[encIndex].referralStatusObs = targetedElem
+              ? targetedElem.display
+              : "";
+            tempEncList.current[encIndex].referralStatusObsUuid = targetedElem
+              ? response.data.uuid
+              : "";
+            setEncountersList(tempEncList.current);
+          } else {
+            setEncountersList(tempEncList.current);
+            //Need error message
+          }
+        }
+      );
+    }
+  };
+
   return (
-    <table className={styles.table}>
-      <thead>
-        <tr className={`omrs-bold ${styles.tr}`}>
-          <td>Link to patient</td>
-          <td>Patient name</td>
-          <td>pathology request</td>
-          <td>Phone number</td>
-          <td>Sample status</td>
-          <td>Date of Request</td>
-          <td>Referral status</td>
-          <td>Sample drop off?</td>
-        </tr>
-      </thead>
-      <tbody>
-        {encountersList.map((enc) => (
-          // foo.results.find(item => item.id === 2)
-
-          <tr className={styles.tr} key={enc.uuid}>
-            <TableRow encounterInfo={enc} />
-          </tr>
-        ))}
-      </tbody>
-    </table>
+    <div>
+      <div className={styles.filtersContainer}>
+        <select
+          className={styles.dropdown}
+          value={sendingHospital}
+          onChange={(e) => handleSendingHospital(e.target.value)}
+        >
+          <option value="" selected disabled hidden>
+            Sending Hospital
+          </option>
+          <option value=""></option>
+          {listOfHospitals.map((loc) =>
+            userLocation ? (
+              loc.uuid === userLocation ? (
+                <option value={loc.display} key={loc.uuid}>
+                  {loc.display}
+                </option>
+              ) : (
+                ""
```

Comment:
```suggestion
                null
```

Not sure if it makes a difference. Just more React-idiomatic.